Present Proof unit tests are falling

```
 [aries-framework/presentproof/service] 2020/04/14 14:18:21 UTC - presentproof.(*Service).newDIDCommActionMsg.func2 -> ERROR stop: delete transitional payload: error
panic: Fail in goroutine after TestService_HandleInbound/Receive_Request_Presentation_(continue_with_proposal) has completed

goroutine 6 [running]:
testing.(*common).Fail(0xc000295c20)
	/Users/runner/hostedtoolcache/go/1.14.2/x64/src/testing/testing.go:613 +0x1f3
testing.(*common).Errorf(0xc000295c20, 0x180c1fe, 0x3, 0xc000058760, 0x1, 0x1)
	/Users/runner/hostedtoolcache/go/1.14.2/x64/src/testing/testing.go:712 +0x9e
github.com/stretchr/testify/assert.Fail(0xa10c260, 0xc000295c20, 0xc00038e140, 0x13d, 0x0, 0x0, 0x0, 0x3c)
	/Users/runner/go/pkg/mod/github.com/stretchr/testify@v1.4.0/assert/assertions.go:261 +0x36a
github.com/stretchr/testify/assert.Equal(0xa10c260, 0xc000295c20, 0x175d1a0, 0x1a0cc20, 0x175d1a0, 0xc0000585d0, 0x0, 0x0, 0x0, 0xc0000585d0)
	/Users/runner/go/pkg/mod/github.com/stretchr/testify@v1.4.0/assert/assertions.go:345 +0x468
github.com/stretchr/testify/require.Equal(0x1a1c660, 0xc000295c20, 0x175d1a0, 0x1a0cc20, 0x175d1a0, 0xc0000585d0, 0x0, 0x0, 0x0)
	/Users/runner/go/pkg/mod/github.com/stretchr/testify@v1.4.0/require/require.go:160 +0xf9
github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/presentproof.TestService_HandleInbound.func10.1(0xc00002a2d0, 0x24, 0xc000382420, 0x0, 0x0)
	/Users/runner/runners/2.168.0/work/aries-framework-go/aries-framework-go/pkg/didcomm/protocol/presentproof/service_test.go:447 +0x17a
reflect.Value.call(0x1775ca0, 0xc000286ee0, 0x13, 0x180c3c8, 0x4, 0xc0003824e0, 0x2, 0x2, 0x0, 0x0, ...)
```

--- a/pkg/didcomm/protocol/presentproof/service_test.go
+++ b/pkg/didcomm/protocol/presentproof/service_test.go
@@ -309,6 +309,8 @@ func TestService_HandleInbound(t *testing.T) {
 
 		messenger.EXPECT().ReplyTo(gomock.Any(), gomock.Any()).
 			Do(func(_ string, msg service.DIDCommMsgMap) error {
+				defer close(done)
+
 				r := &Presentation{}
 				require.NoError(t, msg.Decode(r))
 				require.Equal(t, PresentationMsgType, r.Type)
@@ -325,8 +327,6 @@ func TestService_HandleInbound(t *testing.T) {
 		})
 		store.EXPECT().Delete(gomock.Any()).Return(nil)
 		store.EXPECT().Put(gomock.Any(), gomock.Any()).Do(func(_ string, name []byte) error {
-			defer close(done)
-
 			require.Equal(t, "presentation-sent", string(name))
 
 			return nil
@@ -442,6 +442,8 @@ func TestService_HandleInbound(t *testing.T) {
 
 		messenger.EXPECT().ReplyTo(gomock.Any(), gomock.Any()).
 			Do(func(_ string, msg service.DIDCommMsgMap) error {
+				defer close(done)
+
 				r := &ProposePresentation{}
 				require.NoError(t, msg.Decode(r))
 				require.Equal(t, ProposePresentationMsgType, r.Type)
@@ -458,8 +460,6 @@ func TestService_HandleInbound(t *testing.T) {
 		})
 		store.EXPECT().Delete(gomock.Any()).Return(nil)
 		store.EXPECT().Put(gomock.Any(), gomock.Any()).Do(func(_ string, name []byte) error {
-			defer close(done)
-
 			require.Equal(t, "proposal-sent", string(name))
 
 			return nil
@@ -489,6 +489,8 @@ func TestService_HandleInbound(t *testing.T) {
 
 		messenger.EXPECT().ReplyTo(gomock.Any(), gomock.Any()).
 			Do(func(_ string, msg service.DIDCommMsgMap) error {
+				defer close(done)
+
 				r := &RequestPresentation{}
 				require.NoError(t, msg.Decode(r))
 				require.Equal(t, RequestPresentationMsgType, r.Type)
@@ -506,8 +508,6 @@ func TestService_HandleInbound(t *testing.T) {
 		})
 
 		store.EXPECT().Put(gomock.Any(), gomock.Any()).Do(func(_ string, name []byte) error {
-			defer close(done)
-
 			require.Equal(t, "request-sent", string(name))
 
 			return nil
@@ -669,8 +669,6 @@ func TestService_HandleInbound(t *testing.T) {
 		var done = make(chan struct{})
 
 		store.EXPECT().Put(gomock.Any(), gomock.Any()).Do(func(_ string, name []byte) error {
-			defer close(done)
-
 			require.Equal(t, "request-sent", string(name))
 
 			return nil
@@ -685,6 +683,8 @@ func TestService_HandleInbound(t *testing.T) {
 
 		messenger.EXPECT().Send(msg, Alice, Bob).
 			Do(func(msg service.DIDCommMsgMap, myDID, theirDID string) error {
+				defer close(done)
+
 				return nil
 			})
 
@@ -719,8 +719,6 @@ func TestService_HandleInbound(t *testing.T) {
 		var done = make(chan struct{})
 
 		store.EXPECT().Put(gomock.Any(), gomock.Any()).Do(func(_ string, name []byte) error {
-			defer close(done)
-
 			require.Equal(t, "proposal-sent", string(name))
 
 			return nil
@@ -735,6 +733,8 @@ func TestService_HandleInbound(t *testing.T) {
 
 		messenger.EXPECT().Send(msg, Alice, Bob).
 			Do(func(msg service.DIDCommMsgMap, myDID, theirDID string) error {
+				defer close(done)
+
 				return nil
 			})
 

